api/src/main/java/uk/gov/hmcts/payment/api/service/PaymentRefundsServiceImpl.java

```diff
@@ -203,6 +197,110 @@ public ResponseEntity updateTheRemissionAmount(String paymentReference, Resubmit
         return new ResponseEntity<>(null, HttpStatus.OK);
     }
 
+    @Override
+    public PaymentGroupResponse checkRefundAgainstRemission(MultiValueMap<String, String> headers,
+                                                            PaymentGroupResponse paymentGroupResponse, String ccdCaseNumber) {
+
+        RefundListDtoResponse refundListDtoResponse = getRefundsFromRefundService(ccdCaseNumber, headers);
+
+        var lambdaContext = new Object() {
+            BigDecimal refundAmount = BigDecimal.ZERO;
+        };
+
+        if(paymentGroupResponse.getPaymentGroups()!=null){
+
+            paymentGroupResponse.getPaymentGroups().forEach(paymentGroup ->{
+
+                paymentGroup.getRemissions().forEach(remission -> {
+
+                    remission.setAddRefund(true);
+
+                    remission.setIssueRefund(false);
+
+                    refundListDtoResponse.getRefundList().forEach(refundDto -> {
+
+                        if (refundDto.getRefundReference()!=null && !refundDto.getRefundReference().isBlank()){
+
+                            remission.setAddRefund(false);
+
+                            remission.setIssueRefund(true);
+
+                        }
+                    });
+                });
+
+                if(paymentGroup.getPayments()!=null){
+
+                    paymentGroup.getPayments().forEach(paymentDto -> {
+
+                        refundListDtoResponse.getRefundList().forEach(refundDto -> {
+
+                            lambdaContext.refundAmount = lambdaContext.refundAmount.add(refundDto.getAmount());
+
+                        });
+
+                        if(paymentDto.getAmount().subtract(lambdaContext.refundAmount).compareTo(BigDecimal.ZERO)==1)
+                            paymentDto.setIssueRefundAddRefundAddRemission(true);
+
+                        else
+                            paymentDto.setIssueRefundAddRefundAddRemission(false);
+
+                    });
+                }
+            });
+        }
+
+        return paymentGroupResponse;
+    }
+
+    private RefundListDtoResponse getRefundsFromRefundService(String ccdCaseNumber, MultiValueMap<String, String> headers) {
+
+
+        RefundListDtoResponse refundListDtoResponse;
+
+        UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(refundApiUrl + REFUND_ENDPOINT).queryParam("ccdCaseNumber",ccdCaseNumber);
+
+        LOG.debug("builder.toUriString() : {}", builder.toUriString());
```

Comment:
You should if log.isDebugEnabled() and if true , then add the log.debug statement

Reply:
> You should if log.isDebugEnabled() and if true , then add the log.debug statement

taken care